### PR TITLE
Add intellisense configuration for VSCode

### PR DESCRIPTION
**Pull request by kitbdev**
- closes https://github.com/godotengine/godot-docs/issues/7298
- closes https://github.com/godotengine/godot-docs/issues/8307
- closes https://github.com/godotengine/godot-docs/issues/7218

Added a Configuring Intellisense section for VSCode to address common issues with include errors, such as:
`# include errors detected. cannot open source file "alloca.h"`

Only added examples for Windows, since as far as I know its only a problem on Windows.

There are two ways to fix the issue users have (as far as I know):
1. Setting the `includePath`, `defines`, and `compilerPath` setting in `c_cpp_properties.json`.
2. Using `compiledb=yes` scons arg and  setting the `compileCommands` setting in `c_cpp_properties.json`.

The first way is preferred since if any files are added or changed (including when switching branches in some cases) you will need to recompile to recreate the `compile_commands.json` file.

Added image for the commands to edit the file, since the page has similar. Those commands are important since they can generate the file if it does not already exist.

### CODE REVIEW

**Review by skyace65 — CHANGES_REQUESTED** [review]
Please change the image you added from a png to a webp. I'd recommend converting it with [squoosh](https://squoosh.app/) online, and please keep the image lossless. Also when updating this PR please keep everything to 1 commit, if you don't know how to do that we have a guide [here](https://docs.godotengine.org/en/stable/contributing/workflow/pr_workflow.html#modifying-a-pull-request).

**Review by skyace65 — COMMENTED** [review]
Thanks for updating the image!

**Review by skyace65 — APPROVED** [review]

**Review by mhilbrunner — APPROVED** [review]

File: contributing/development/configuring_an_ide/visual_studio_code.rst [hunk 1]
```diff
@@ -156,6 +156,60 @@ The name under ``program`` depends on your build configuration,
 e.g. ``godot.linuxbsd.editor.dev.x86_64`` for 64-bit LinuxBSD platform with
 ``platform=editor`` and ``dev_build=yes``.
 
+Configuring Intellisense
+---------------------
```
Comment (mhilbrunner) [inline contributing/development/configuring_an_ide/visual_studio_code.rst:160]:
```suggestion
Configuring Intellisense
------------------------
```

**Comment by mhilbrunner** [comment]
Merged. Thanks and congrats on your first merged contribution!

**Comment by mhilbrunner** [comment]
Cherry-picked to 4.2 in https://github.com/godotengine/godot-docs/pull/9648.

**Pull request merged**